src/Gzero/Core/Query/Condition.php

```diff
@@ -181,20 +187,39 @@ protected function validate()
         if (!in_array($this->operation, self::$allowedOperations, true)) {
             throw new InvalidArgumentException('Unsupported condition operation');
         }
-        if (is_array($this->value) && !$this->isCorrectRangeFormat()) {
+
+        if ($this->isArrayOperation() && !is_array($this->value)) {
+            throw new InvalidArgumentException('Value is not of type array');
+        }
+
+        if (is_array($this->value) && $this->isNotCorrectRangeFormat()) {
             throw new InvalidArgumentException('Wrong number of values for range');
         }
     }
 
     /**
-     * Checks if it's correct range format
+     * Checks if it is not correct range format
+     *
+     * @return bool
+     */
+    protected function isNotCorrectRangeFormat(): bool
```

Comment:
Why is not just named: `isCorrectRangeFormat`? Just alter the inside logic to conform to such name ;)

Reply:
:+1: